core/src/main/kotlin/xtdb/buffer_pool/RemoteBufferPool.kt

```diff
@@ -177,7 +179,11 @@ class RemoteBufferPool(
             diskCache.get(key) { k, tmpFile ->
                 diskCacheMisses.increment()
                 objectStore.getObject(k, tmpFile)
-            }.thenApply { entry -> Pair(PathSlice(entry.path, pathSlice.offset, pathSlice.length), entry) }
+                      }.thenApply {
+                          entry ->
+                          networkRead.increment(entry.path.fileSize().toDouble())
```

Comment:
This is not the correct location. You are counting it whenever we fetch something from disk. You need to apply it at the same level as the `diskCacheMisses`.